src/components/flame-graph/Canvas.js

```diff
@@ -193,19 +251,45 @@ class FlameGraphCanvasImpl extends React.PureComponent<Props> {
         continue;
       }
 
+      const cssRowTop: CssPixels =
+        (maxStackDepth - depth - 1) * ROW_HEIGHT - viewportTop;
+      const cssRowBottom: CssPixels =
+        (maxStackDepth - depth) * ROW_HEIGHT - viewportTop;
+      const deviceRowTop: DevicePixels = snap(cssRowTop * cssToDeviceScale);
+      const deviceRowBottom: DevicePixels =
+        snap(cssRowBottom * cssToDeviceScale) - 1;
+      const deviceRowHeight: DevicePixels = deviceRowBottom - deviceRowTop;
+
+      const deviceTextTop =
+        deviceRowTop + snap(TEXT_OFFSET_TOP * cssToDeviceScale);
+
       for (let i = 0; i < stackTiming.length; i++) {
-        const startTime = stackTiming.start[i];
-        const endTime = stackTiming.end[i];
+        // For each box, snap the left and right edges to the nearest multiple
+        // of two device pixels. If both edges snap to the same value, the box
+        // becomes empty and is not drawn.
```

Comment:
I'm not sure of this. I think we should snap to at least 1px if there's a box, but do it just for one such small box on this px (if there are more than one box here), and only if there's no bigger box at this location.
This would be similar to (but not completely the same) what we do in the marker chart.
What do you think?

Reply:
Currently, when selecting a preview while viewing the flame graph on your deploy preview, we see "peaks" showing up and then disappearing very often. I don't know if this is the reason, but I'd expect the peaks to still show up even if they have a small width. It would be great if the view would stay mostly stable.